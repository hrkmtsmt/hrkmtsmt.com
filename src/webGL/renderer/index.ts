import { MutableRefObject } from 'react';
import {
  BoxGeometry,
  Color,
  DirectionalLight,
  Mesh,
  MeshStandardMaterial,
  PerspectiveCamera,
  Scene,
  WebGL1Renderer
} from 'three';

export const threeRenderer = (ref: MutableRefObject<HTMLDivElement | null>) => {
  const renderer = new WebGL1Renderer();

  renderer.domElement.classList.add('w-full');
  renderer.setPixelRatio(window.devicePixelRatio);
  renderer.setClearColor(0x000000, 0);
  ref.current.appendChild(renderer.domElement);

  const scene = new Scene();

  const camera = new PerspectiveCamera(45, 1, 1, 10000);
  camera.position.set(0, 0, 1000);

  const geometry = new BoxGeometry(200, 200, 200);

  const primaryColor = new Color('rgb(47, 118, 230)');

  const material = new MeshStandardMaterial({ color: primaryColor });

  const boxMesh = new Mesh(geometry, material);
  scene.add(boxMesh);

  const light = new DirectionalLight(0xffffff);
  light.position.set(1, 1, 1);
  scene.add(light);

  tick();

  function tick() {
    requestAnimationFrame(tick);

    boxMesh.rotation.x += 0.01;
    boxMesh.rotation.y += 0.01;

    renderer.render(scene, camera);
  }

  return () => {
    const emptyElement = document.createElement('div');
    ref.current.appendChild(emptyElement);
  };
};
