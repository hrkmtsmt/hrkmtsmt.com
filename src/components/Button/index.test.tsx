import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Button } from '.';

describe('Button', () => {
  it('Snapshot', () => {
    const snapshot = renderer.create(<Button color={'primary'}>Button</Button>);
    expect(snapshot).toMatchSnapshot();
  });

  it('Checking role and text', () => {
    render(<Button color={'primary'}>Button</Button>);
    expect(screen.getByRole('button', { name: /Button/ })).toBeDefined();
  });

  it('Checking color primary', () => {
    const { container } = render(<Button color={'primary'}>Button</Button>);
    expect(container.querySelectorAll('bg-primary')).toBeDefined();
  });

  it('Checking color secondary', () => {
    const { container } = render(<Button color={'secondary'}>Button</Button>);
    expect(container.querySelectorAll('bg-secondary')).toBeDefined();
  });
});
