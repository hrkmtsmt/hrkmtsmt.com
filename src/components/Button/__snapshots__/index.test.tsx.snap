// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot 1`] = `
<button
  className="bg-primary text-white before:bg-secondary hover:bg-secondary hover:text-primary relative z-10 flex w-fit overflow-hidden rounded-full px-5 text-center font-bold leading-10 duration-200 ease-in-out before:absolute before:-left-0 before:-z-10 before:h-10 before:w-full before:-translate-x-full before:rounded-full before:duration-200 before:ease-in-out before:content-[\\"\\"] hover:duration-200 hover:ease-in-out hover:before:translate-x-0 active:scale-95"
  color="primary"
>
  Button
</button>
`;
